WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/logic/interval/intervaltypes/IntervalBase.java

```diff
@@ -122,7 +122,11 @@ public IntervalType getType() {
 
 	/** Necessary for storage of Intervals. */
 	public int compareTo(IntervalBase comparedInterval) {
-		return getEnd().compareTo(comparedInterval.getEnd());
+		int res = getEnd().compareTo(comparedInterval.getEnd());
+		if (res == 0 && !this.equals(comparedInterval)) {
+			res = -1;
+		}
+		return res;
```

Comment:
My assumption was not complete: Assume compareTo(a,b)=-1 and compareTo(a.end,b.end)=0 (which is the error scenario you are fixing here). Then we still have the above problem, IMHO.